### PR TITLE
68 display items

### LINKED ISSUE #22: Visualizing Items on the map
As a Buyer, I want to be able to visualize the items around me, to get a better sense of what offers are easy to pick up

### PR DESCRIPTION
**Pull request by v1ruso**
Fix #22, #68 

I added a way to see the items on the map (with markers in blue), that can be clicked to obtain a description of the item at that location. Only the items that aren't "sold" are displayed.

I also added a button that resets the map center and zoom.

Screenshot 2021-04-10 at 20 40 33

Screenshot 2021-04-10 at 20 41 22

### CODE REVIEW

File: app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt [hunk 1]
```diff
@@ -33,19 +33,41 @@ class MapFragmentTest {
...
     @Test
     fun itemsAddedAreDisplayedOnTheMap() {
         navigate_to(R.id.newItemFragment)
+        onView(withId(R.id.editItemTitle)).perform(
+            ViewActions.typeText(itemTitle),
+            ViewActions.closeSoftKeyboard()
+        )
         onView(withId(R.id.new_item_get_location)).perform(click())
-        Thread.sleep(3000)
+        Thread.sleep(6000)
```
Comment (cronokirby) [inline app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt:47]: Maybe we could make a constant for the sleep time, that way it's easier to configure

File: app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt [hunk 2]
```diff
@@ -33,19 +33,41 @@ class MapFragmentTest {
...
+        var result = 0
+        val resourceId: Int =
+            activity!!.resources.getIdentifier("status_bar_height", "dimen", "android")
+        if (resourceId > 0) {
+            result = activity!!.resources.getDimensionPixelSize(resourceId)
+        }
         val device = UiDevice.getInstance(getInstrumentation())
-        val marker = device.findObject(UiSelector().descriptionContains(""))
-        marker.click()
+        device.click(device.displayWidth / 2, device.displayHeight / 2 + result)
```
Comment (cronokirby) [inline app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt:62]: This seems like an error prone method, but I guess there's not really any alternative to working with the map is there?

If this causes even flakier CI, we're going to have fun later.

File: app/src/androidTest/java/com/example/sharingang/DetailedItemFragmentTest.kt [hunk 3]
```diff
@@ -25,7 +25,7 @@ class DetailedItemFragmentTest {
 
     private val itemTitle = "T"
     private val itemDescription = "D"
-
+/*
```
Comment (cronokirby) [inline app/src/androidTest/java/com/example/sharingang/DetailedItemFragmentTest.kt:28]: We should just remove the code instead of commenting it out

**Review by cronokirby — APPROVED** [review]
Code looks good to me; the test looks a bit flaky to me, but I assume you know more about getting it to work on the CI than I do.

There's also a bit of code that's commented out, which we should remove, but it's not worth wrangling with the CI over. If there's some other change worthy, might be a good idea to include it.

File: app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt (hunk 2 above)
Comment (v1ruso) [reply app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt:62]: Unfortunately there is no way to directly click on a specific marker on the map. Instead, I decided to place an object at "my" location, then wait for the map to center itself around "my" location, and then I click at the center of the screen, taking into account the status bar width.

File: app/src/androidTest/java/com/example/sharingang/DetailedItemFragmentTest.kt (hunk 3 above)
Comment (v1ruso) [reply app/src/androidTest/java/com/example/sharingang/DetailedItemFragmentTest.kt:28]: I thought maybe we would want to reuse it later?

Comment (cronokirby) [reply app/src/androidTest/java/com/example/sharingang/DetailedItemFragmentTest.kt]: It's still in the history of the file

Comment (v1ruso) [reply app/src/androidTest/java/com/example/sharingang/DetailedItemFragmentTest.kt]: You're right

**Review by lumapools — APPROVED** [review]
Looks good to me. I have no additional remarks to what Lucas said, my questions have been answered looking at the comments. Well done!

**Pull request merged**